### PR TITLE
feat(transformer/typescript): reports error for import lib = require(...);

### PR DESCRIPTION
**Pull request by Dunqing**

### CODE REVIEW

**Comment by Dunqing** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3020?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3021** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3021?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3020** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3020?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3018** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3018?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by Boshen — APPROVED** [review]

**Comment by Boshen** [comment]
### Merge activity

* **Apr 18, 6:52 AM EDT**: @Boshen started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/oxc-project/oxc/3020).

File: crates/oxc_transformer/src/typescript/diagnostics.rs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_span::Span;
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("`import lib = require(...);` is only supported when compiling modules to CommonJS.\nPlease consider using `import lib from '...';` alongside Typescript's --allowSyntheticDefaultImports option, or add @babel/plugin-transform-modules-commonjs to your Babel config.")]
```
Comment (milesj) [inline crates/oxc_transformer/src/typescript/diagnostics.rs:8]: @Dunqing @Boshen do we want to reference babel here?

Comment (Dunqing) [reply crates/oxc_transformer/src/typescript/diagnostics.rs:8]: I think that until oxc has the modules-commonjs plugin, we can make its errors consistent with babel for the time being

**Pull request merged**